cyvlfeat/sift/phow.py

```diff
@@ -0,0 +1,96 @@
+import math
+import numpy as np
+import matplotlib.colors as colors
+import scipy.ndimage
+from . import dsift
+
+
+def phow(image, verbose=False, fast=True, sizes=(4, 6, 8, 10), step=2, color='gray',
+         float_descriptors=False, magnif=6, window_size=1.5, contrast_threshold=0.005):
+    global frames, descrs
+
+    # Extract the features
+    def rgb2gray(rgb):
+        return np.dot(rgb[..., :3], [0.299, 0.587, 0.114])
+
+    # standardize the image
+    color_lower = color.lower()
+    set1 = set(color_lower.split(''))
+    set2 = set('gray'.split(' '))
+    if set1 == set2:
+        num_channels = 1
+
+    if image.shape[2] > 1:
+        image = rgb2gray(image)
```

Comment:
This will always make the image grey, which negates the point of the block below which does other colorspace conversion.